### PR TITLE
Add FSMRMP font package

### PR DESCRIPTION
**Pull request by digitalrane**
## Description of changes

This adds a new font package, FSMRMP, a patched font combining [Fantasque Sans Mono](https://fontlibrary.org/en/font/fantasque-sans-mono), [Rounded Mgen plus](http://jikasei.me/font/rounded-mgenplus/), [Noto Emoji](https://fonts.google.com/noto/specimen/Noto+Emoji) (Monochrome) and [isfit+](https://github.com/uwabami/isfit-plus) icons. It is a font similar in style to Fantasque Sans Mono but also useful for reading & writing Japanese.

I have replicated the functionality of the project Makefile in the Nix packaging definition so that all required font sources can be downloaded prior to build - the original Makefile downloads them at build time which doesn't work with sandboxing.

This package builds the font at package build time as the project does not release compiled font files which could otherwise be downloaded. I've done a test build on x86_64 under NixOS and the package builds - however this is my first contribution to Nix so I'm still learning and any help and guidance around where I could improve would be very much appreciated 🙏🏻 

The GitHub project is [here](https://github.com/uwabami/fsmrmp) for reference.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/data/fonts/fsmrmp/default.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+, fetchgit
+, fetchurl
+, pkgs
+}:
+
+stdenv.mkDerivation rec {
+  pname = "fsmrmp";
+  version = "2023-08-24";
+
+  src = fetchgit {
```
Comment (WolfangAukang) [inline pkgs/data/fonts/fsmrmp/default.nix:12]: Please use `fetchFromGitHub` instead

File: pkgs/data/fonts/fsmrmp/default.nix [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    hash = "sha256-a7OyRBO3ju0Z/6m9IzrlVZguOxhb0wPlfdHgW+vxc1I=";
+  };
+  notosrc = fetchurl {
+    url = "https://fonts.google.com/download?family=Noto%20Emoji";
+    hash = "sha256-nYy685h/gd3G66k50Ts5CFjs4De8CgJWYiSnGfeXJvE=";
+  };
+  isfitsrc = fetchurl {
+    url = "https://github.com/uwabami/isfit-plus/raw/master/dists/isfit-plus.ttf";
+    hash = "sha256-evL7bpTxTsReyrhKMqdp/fe4bus1EEVDi2FheyZzRlw=";
+  };
```
Comment (WolfangAukang) [inline pkgs/data/fonts/fsmrmp/default.nix:32]: Move this to a `let` section instead

File: pkgs/data/fonts/fsmrmp/default.nix [hunk 3]
```diff
@@ -0,0 +1,64 @@
+{ lib
+, stdenv
+, fetchgit
+, fetchurl
+, pkgs
```
Comment (WolfangAukang) [inline pkgs/data/fonts/fsmrmp/default.nix:5]:
```suggestion
, python3
, fontforge
, unar
```
Do not use `pkgs` as a parameter. This is already being called by `callPackage`, so you will only need to add the needed packages.

File: pkgs/data/fonts/fsmrmp/default.nix [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  notosrc = fetchurl {
+    url = "https://fonts.google.com/download?family=Noto%20Emoji";
+    hash = "sha256-nYy685h/gd3G66k50Ts5CFjs4De8CgJWYiSnGfeXJvE=";
+  };
+  isfitsrc = fetchurl {
+    url = "https://github.com/uwabami/isfit-plus/raw/master/dists/isfit-plus.ttf";
+    hash = "sha256-evL7bpTxTsReyrhKMqdp/fe4bus1EEVDi2FheyZzRlw=";
+  };
+
+  nativeBuildInputs = ([pkgs.python3 pkgs.fontforge pkgs.unar ]);
```
Comment (WolfangAukang) [inline pkgs/data/fonts/fsmrmp/default.nix:34]:
```suggestion
  nativeBuildInputs = [
    python3
    fontforge
    unar
  ];
```

**Review by WolfangAukang — CHANGES_REQUESTED** [review]

**Comment by digitalrayne** [comment]
@WolfangAukang - thank you for the review, and sorry it took me a while to get back to addressing your feedback. I've been moving my machines over to NixOS so my main development machine was out of commission :)

I've rebased this, and addressed your feedback. I've run a test build and the package is building correctly. I've also moved the package under `by-name` too in keeping with newer conventions.

I've also updated the hash of the Noto font, as it changed during testing. I'm wondering if this is OK, or if I need to rethink how the font is being fetched to prevent build failures in the future due to the font updating and no longer matching the hash in the package?

Thanks again for your time reviewing this.

**Pull request closed without merging**